rskj-core/src/main/java/org/ethereum/db/MutableRepository.java

```diff
@@ -299,6 +299,10 @@ public synchronized Coin getBalance(RskAddress addr) {
     public synchronized Coin addBalance(RskAddress addr, Coin value) {
         AccountState account = getAccountStateOrCreateNew(addr);
 
+        if (value.equals(new Coin(BigInteger.ZERO))) {
+            return account.getBalance();
+        }
+
         Coin result = account.addToBalance(value);
         updateAccountState(addr, account);
```

Comment:
Another possibility is:

```suggestion
        Coin result = account.addToBalance(value);

        if (!value.equals(new Coin(BigInteger.ZERO))) updateAccountState(addr, account);
```